### PR TITLE
MYSQL-7 : Password Values not redacted in the shell logs , MYSQL-18 Environment refresh and discovery operation shows passwords/sensitive data

### PR DESCRIPTION
**Pull request by vinaybyrappa**
Background :-
Passwords values not redacted in the shell logs , Environment refresh and discovery operation shows passwords/sensitive data

Solution 
All the passwords are masked in the shell and in the python files all the successfull messages which contains information of password is removed 
VDB title was showing password and other db credentials, have changed that to show only db name

### CODE REVIEW

File: src/resources/restore_stage.sh [hunk 1]
```diff
@@ -101,15 +100,13 @@ ${MYSQLD}/mysqld --initialize --user=mysql --datadir=${NEW_DATA_DIR} --log-error
 PWD_LINE=`cat ${NEW_DATA_DIR}/mysqld.log | grep 'temporary password'`
 # sudo grep 'temporary password' ${NEW_DATA_DIR}/mysqld.log`
 # 2019-04-11T14:40:34.032576Z 1 [Note] A temporary password is generated for root@localhost: L0qXNZ8?C3Us
-log "init temporary password: ${PWD_LINE}"
 
 TMP_PWD=`echo "${PWD_LINE}" | ${AWK} -F": " '{print $2}' | xargs`
 # These temporary passwords contain special characters so need to wrap in single / literal quotes ...
 TMP_PWD=`echo "'"$TMP_PWD"'"`
-log "Temporary Password: ${TMP_PWD}"
+# log "Temporary Password: ${TMP_PWD}"
```
Comment (gmanhas23) [inline src/resources/restore_stage.sh:107]: Instead of commenting, could we just remove this? here and everywhere else

**Review by gmanhas23 — APPROVED** [review]
Looks good to me... Just the one commented line that we need to remove

File: src/resources/restore_stage.sh (hunk 1 above)
Comment (vinaybyrappa) [reply src/resources/restore_stage.sh:107]: I have removed it whereever there was it was commented before . I have removed all the log statements

File: src/pluginops/pluginops.py [hunk 2]
```diff
@@ -576,12 +568,11 @@ def configure(virtual_source, snapshot, repository):
     }
     configure_script = pkgutil.get_data('resources', 'provision.sh')
     result = libs.run_bash(virtual_source.connection, configure_script,environment_vars,check=False)
-    logger.debug(result)
     output = result.stdout.strip()
     std_err=result.stderr.strip()
     exit_code = result.exit_code
     if exit_code == 0:
-        logger.debug("Pre-Snapshot/Restore_DB successful "+output)
+        logger.debug("Pre-Snapshot/Restore_DB successful"+output)
```
Comment (batrankit) [inline src/pluginops/pluginops.py:575]: Not sure if the intention here was to remove the output. Please check.

File: src/resources/provision.sh [hunk 3]
```diff
@@ -216,7 +216,7 @@ outputJSON=$($DLPX_BIN_JQ ".baseDir = $(jqQuote "$CONFIG_BASEDIR")" <<< "$output
 outputJSON=$($DLPX_BIN_JQ ".dbName = $(jqQuote "$prettyName")" <<< "$outputJSON")
 
 printf "$outputJSON" > "$DLPX_OUTPUT_FILE"
-log "Output: $outputJSON"
+#log "Output: $outputJSON"
```
Comment (batrankit) [inline src/resources/provision.sh:219]: Please remove this instead of commenting.

File: src/resources/restore_stage.sh (hunk 1 above)
Comment (batrankit) [reply src/resources/restore_stage.sh:107]: This is still being seen in the code as being commented.

File: src/resources/restore_stage_bi.sh [hunk 4]
```diff
@@ -75,15 +74,13 @@ ${MYSQLD}/mysqld --initialize --user=mysql --datadir=${NEW_DATA_DIR} --log-error
 
 PWD_LINE=`cat ${NEW_DATA_DIR}/mysqld.log | grep 'temporary password'`
 # sudo grep 'temporary password' ${NEW_DATA_DIR}/mysqld.log`
-log "init temporary password: ${PWD_LINE}"
+# log "init temporary password: ${PWD_LINE}"
```
Comment (batrankit) [inline src/resources/restore_stage_bi.sh:77]: Same, Can we remove this instead of commenting.

**Review by batrankit — COMMENTED** [review]
**Comment 1:-**
Some of these files have the copyright statements and some do not. 

We need to make it consistent. Also where ever copyright is present, that has not been updated.

**Comment 2:-**
Can you please check if the password is still being seen at the below places from the logs:-
delphix_mysql_debug.log:[2023-01-16T03:54:21][DEBUG][restore_stage_bi.sh][]:[Creation Results: { "conn": "-uroot --host=localhost -p***** "pass": "'?I>;bjcZ[2023-01-16T03:54:21][DEBUG][restore_stage_bi.sh][]:[Creating Staging Directories on NFS Mounted Path from Delphix ...]

delphix_mysql_debug.log:[2023-01-16T03:54:21][DEBUG][restore_stage_bi.sh][]:[Return message:mysql: [Warning] Using a password on the command line interface can be insecure.]
delphix_mysql_debug.log:SOURCEPASS=Pass@123
delphix_mysql_debug.log:STAGINGPASS=



[2023-01-16T04:34:26][DEBUG][restore_stage.sh][]:[Creation Results: {
    "conn": "-uroot --host=localhost -p",
    "pass": "'Func8=>sTqhU'",
    "string": "-uroot --host=localhost -p'Func8=>sTqhU'  --protocol=TCP --port=3308"
    }]

File: src/pluginops/pluginops.py (hunk 2 above)
Comment (vinaybyrappa) [reply src/pluginops/pluginops.py:575]: The output contains password as well , so had to remove this line

**Review by gmanhas23 — CHANGES_REQUESTED** [review]
Line 477 ( now 474) in `restore_stage_bi.sh` is logging SOURCEPASS in debug  logs
Screenshot 2023-01-17 at 5 35 40 PM

File: src/resources/restore_stage_bi.sh [hunk 5]
```diff
@@ -474,7 +467,7 @@ command_runner "${CMD}" 11
 log "Environment:"
 export DLPX_LIBRARY_SOURCE=""
 export STAGINGPASS=""
-env | sort  >>$DEBUG_LOG
+env | grep -v 'STAGINGPASS' | sort >>$DEBUG_LOG
```
Comment (gmanhas23) [inline src/resources/restore_stage_bi.sh:470]: The issue is the SOURCEPASS being shown in logs with env... maybe we can remove all occurrences of PASS from the command, like so 
`env | grep -v 'STAGINGPASS' | sort >>$DEBUG_LOG`

File: src/pluginops/pluginops.py [hunk 6]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2020 by Delphix. All rights reserved.
+# Copyright (c) 2021, 2023 by Delphix. All rights reserved.
```
Comment (batrankit) [inline src/pluginops/pluginops.py:2]: If any file is having a copyright from any previous year than 2021, Please don't change that it to 2021. So in this case, you can use # Copyright (c) 2020, 2023 by Delphix. All rights reserved.

File: src/resources/library.sh [hunk 7]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2020 by Delphix. All rights reserved.
+# Copyright (c) 2021, 2023 by Delphix. All rights reserved.
```
Comment (batrankit) [inline src/resources/library.sh:2]: Same as above.

File: src/resources/provision.sh [hunk 8]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2018 by Delphix. All rights reserved.
+# Copyright (c) 2021, 2023 by Delphix. All rights reserved.
```
Comment (batrankit) [inline src/resources/provision.sh:2]: Same as above.

File: src/resources/restore_stage.sh [hunk 9]
```diff
@@ -588,6 +583,6 @@ export DLPX_LIBRARY_SOURCE=""
 export REPLICATION_PASS=""
 export STAGINGPASS=""
 export SOURCEPASS=""
-env | sort  >>$DEBUG_LOG
+env | grep -v 'PASS' | sort >>$DEBUG_LOG
```
Comment (batrankit) [inline src/resources/restore_stage.sh:586]: Do we actually need this information in the log file. Can we remove this file entirely?

File: src/resources/restore_stage_bi.sh [hunk 10]
```diff
@@ -474,7 +467,7 @@ command_runner "${CMD}" 11
 log "Environment:"
 export DLPX_LIBRARY_SOURCE=""
 export STAGINGPASS=""
-env | sort  >>$DEBUG_LOG
+env | grep -v 'PASS' | sort >>$DEBUG_LOG
```
Comment (batrankit) [inline src/resources/restore_stage_bi.sh:470]: Same as above.

File: src/resources/restore_stage_si.sh [hunk 11]
```diff
@@ -263,8 +261,6 @@ eval ${CMD} 1>>${DEBUG_LOG} 2>&1
 
 log "Environment: "
 export DLPX_LIBRARY_SOURCE=""
-#export REPLICATION_PASS=""
-#export STAGINGPASS=""
-env | sort  >>$DEBUG_LOG
+env | grep -v 'PASS' | sort >>$DEBUG_LOG
```
Comment (batrankit) [inline src/resources/restore_stage_si.sh:264]: Same as above.

**Review by batrankit — APPROVED** [review]

**Review by gmanhas23 — APPROVED** [review]

**Pull request merged**